src/Orchard/Localization/Services/DefaultLocalizedStringManager.cs

```diff
@@ -118,70 +122,53 @@ public class DefaultLocalizedStringManager : ILocalizedStringManager {
         private IDictionary<string, string> LoadTranslationsForCulture(string culture, AcquireContext<string> context) {
             IDictionary<string, string> translations = new Dictionary<string, string>();
             string corePath = string.Format(CoreLocalizationFilePathFormat, culture);
-            string text = _webSiteFolder.ReadFile(corePath);
-            if (text != null) {
-                _localizationStreamParser.ParseLocalizationStream(text, translations, false);
-                if (!DisableMonitoring) {
-                    Logger.Debug("Monitoring virtual path \"{0}\"", corePath);
-                    context.Monitor(_webSiteFolder.WhenPathChanges(corePath));
-                }
-            }
+            LoadTranslationsFromLanguageFilePath(corePath, translations, context, false);
 
             foreach (var module in _extensionManager.AvailableExtensions()) {
                 if (DefaultExtensionTypes.IsModule(module.ExtensionType)) {
                     string modulePath = string.Format(ModulesLocalizationFilePathFormat, module.VirtualPath, culture);
-                    text = _webSiteFolder.ReadFile(modulePath);
-                    if (text != null) {
-                        _localizationStreamParser.ParseLocalizationStream(text, translations, true);
-
-                        if (!DisableMonitoring) {
-                            Logger.Debug("Monitoring virtual path \"{0}\"", modulePath);
-                            context.Monitor(_webSiteFolder.WhenPathChanges(modulePath));
-                        }
-                    }
+                    LoadTranslationsFromLanguageFilePath(modulePath, translations, context, true);
                 }
             }
 
-            foreach (var theme in _extensionManager.AvailableExtensions()) {
-                if (DefaultExtensionTypes.IsTheme(theme.ExtensionType) && _shellDescriptor.Features.Any(x => x.Name == theme.Id))
-                {
+            var sitePart = _workContextAccessor.GetContext().CurrentSite as IContent;
+            var themePart = sitePart.ContentItem.Parts.FirstOrDefault(e => e.PartDefinition.Name == "ThemeSiteSettingsPart");
+            var currentThemeName = themePart?.Retrieve<string>("CurrentThemeName");
```

Comment:
Haven't looked in depth at the PR but there shouldn't be any reason to look into the settings directly, it should use the Theme selectors instead, with the ThemeManager.